R/p_direction.R

```diff
@@ -35,28 +40,40 @@ pd <- p_direction
 
 
 
-
+#' @importFrom stats density
 #' @rdname p_direction
 #' @export
-p_direction.numeric <- function(x, ...) {
-  p_direction <- 100 * max(
-    c(
-      length(x[x > 0]) / length(x), # pd positive
-      length(x[x < 0]) / length(x) # pd negative
+p_direction.numeric <- function(x, method = "direct", ...) {
+  if (method == "direct") {
+    p_direction <- 100 * max(
+      c(
+        length(x[x > 0]) / length(x), # pd positive
+        length(x[x < 0]) / length(x) # pd negative
+      )
     )
-  )
-
+  } else {
+    dens <- as.data.frame(stats::density(x, n = 2^10))
```

Comment:
@mattansb Based on what you mentioned about the difference between logspline and base density, would you recommend using the logspline density instead for density here?

Reply:
`logspline`'s algorithm seems to give more sensible results (especially around distribution tails/edges). So if edges/tails are important for AUC calculations, it might be better to use `logspline`. If not, base is probably fine.

Reply:
As the tail is the only thing of importance here, I think it would make sense 😅

Reply:
Maybe we should implement our wrapper for logspline density, since it's used at different places. Something like `estimate_density` or something.

Reply:
(we could even ~steal~ borrow the [logspline](https://github.com/cran/logspline/blob/master/R/logspline.R) function to avoid the dep issue...

Reply:
Unfortunately, there is some compiled C code in the package that it makes it copy non-trivial. Should we exceptionally add logspline to the dependencies (it is quite light)?

Reply:
If we want to keep dependency light, you can default to `stats::density` method if `logspline` is missing, with a message (like in `bayesfactor_savagedickey`):
````
insight::print_color("Consider installing the \"logspline\" package for a more robust estimate.\n", "red")
````